### ISSUE TITLE
Supporting refs for internal and external use

### ISSUE THREAD

**Issue by jamesplease** [jamesplease/materialish#272]
Presently, there is a divide of sorts when it comes to Materialish refs. If a ref is used internally, then it cannot be used externally. And vice versa.

There are only a few situations when refs are needed internally. But those components do not have a `nodeRef` prop.

@conorhastings proposed a solution to this in https://github.com/jamesplease/materialish/pull/267#issuecomment-414174150 (**[CodeSandbox Example](https://codesandbox.io/s/8xk9225jq9)**).

There is a catch: it doesn't support string refs. But that may okay, given the benefits that this API provides, and the fact that string refs are deprecated.

**Comment by conorhastings** [comment]
thinking further about this, not allowing string refs seems especially fine since the prop is nodeRef, they would actually have direct access via `this.refs['string']` they would have to attach a direct ref to the materialish node and use `this.refs['something'].children.refs['string']`

I think calling out lack of support for string refs in nodeRef in docs seems appropriate since above workflow is really poor anyway

**Comment by jamesplease** [comment]
Over in #240 , this would be pretty useful to enable us to refocus the input after the clearable button is clicked.

**Comment by jamesplease** [comment]
Here is what I'm thinking:

```js
getRef = ref => {
  const { nodeRef } = this.props;

  if (typeof nodeRef === 'function') {
    nodeRef(ref);
  } else if (nodeRef && nodeRef.hasProperty('current')) {
    nodeRef.current = ref;
  }

  this.nodeRef = ref;
}
```

**Comment by jamesplease** [comment]
Resolved in #277 . This will be released as 0.17.0; possibly tonight or tomorrow. I'm just doing a bit more testing to make sure that everything is good.
